test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -1215,14 +1215,14 @@ TEST_F(InjaTransformerTest, ReplaceWithRandomTest_SameReplacementPatternUsesSame
 {{{{ replace_with_random("{}", "{}") }}}}
   )ENDFMT";
 
-  auto formatted_string = fmt::format(format_string,
+  auto formatted_string = fmt::vformat(format_string, fmt::make_format_args(
```

Comment:
I dont think we need to do this. Lets just change the test strings and patterns to be std::string rather than auto.

Auto is detecting them as char * which is why I think they arent working by default with format.

Reply:
okay, it took some wrestling with the compiler, but this now works: a706ea8e3907589b97dfc842f286c6fa31f43b56

Reply:
https://github.com/solo-io/envoy-gloo/pull/333/files#r1600010071

Reply:
Did not work unfortunately. Here is the compiler output:

```
ERROR: /build/envoy-gloo/test/extensions/filters/http/transformation/BUILD:16:19: Compiling test/extensions/filters/http/transformation/inja_transformer_test.cc failed: (Exit 1): clang-14 failed: error executing command (from target //test/extensions/filters/http/transformation:inja_transformer_test_lib_internal_only) /opt/llvm/bin/clang-14 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer '-std=c++14' ... (remaining 312 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
test/extensions/filters/http/transformation/inja_transformer_test.cc:1218:39: error: call to consteval function 'fmt::basic_format_string<char, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &, std::basic_string<char> &>::basic_format_string<std::basic_string<char>, 0>' is not a constant expression
  auto formatted_string = fmt::format(format_string,
                                      ^
external/com_github_fmtlib_fmt/include/fmt/core.h:464:17: note: non-constexpr function 'data' cannot be used in a constant expression
      : data_(s.data()), size_(s.size()) {}
                ^
external/com_github_fmtlib_fmt/include/fmt/core.h:3148:62: note: in call to 'basic_string_view(format_string)'
  FMT_CONSTEVAL FMT_INLINE basic_format_string(const S& s) : str_(s) {
                                                             ^
test/extensions/filters/http/transformation/inja_transformer_test.cc:1218:39: note: in call to 'basic_format_string(format_string)'
  auto formatted_string = fmt::format(format_string,
                                      ^
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/basic_string.h:2332:7: note: declared here
      data() const _GLIBCXX_NOEXCEPT
      ^
1 error generated.
```